src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -9258,6 +9258,53 @@ void M(S s)
                 Diagnostic(ErrorCode.ERR_FixedNeeded, "&(s_f.Buf[0])").WithLocation(17, 18));
         }
 
+        [Fact, WorkItem(39632, "https://github.com/dotnet/roslyn/issues/39632")]
+        public void CanUseIndexInitialiserWithFixedSizeBuffer()
+        {
+            var source = @"
+unsafe struct S
+{
+    public fixed double V[3];
+
+    static void M()
+    {
+        var b = new S { V = { [0] = 1 } };
+    }
+}";
+            CreateCompilation(source, options: TestOptions.UnsafeDebugDll)
+                .VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(39632, "https://github.com/dotnet/roslyn/issues/39632")]
+        public void CanNotUseObjectOrCollectionInitialiserWithFixedSizeBufferOutsideOfFixedContext()
+        {
+            var source = @"
+unsafe struct S
+{
+    public fixed double V[3];
+
+    static void M()
+    {
+        var b = new S { V = { 1 } };
+        b = new S { V = { Prop = 1 } };
```

Comment:
>{ Prop = 1 } [](start = 24, length = 12)

Consider also testing with an empty initializer and with an initializer that mixes assignments (indexed followed by non-indexed and in reversed order).